internal/provider/resource_basistheory_proxy.go

```diff
@@ -0,0 +1,183 @@
+package provider
+
+import (
+	"context"
+	"github.com/Basis-Theory/basistheory-go/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceBasisTheoryProxy() *schema.Resource {
+	return &schema.Resource{
+		Description: "Proxy https://docs.basistheory.com/#proxies",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		CreateContext: resourceProxyCreate,
+		ReadContext:   resourceProxyRead,
+		UpdateContext: resourceProxyUpdate,
+		DeleteContext: resourceProxyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "Unique identifier for the Proxy",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"key": {
+				Description: "Key for the Proxy",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"tenant_id": {
+				Description: "Tenant identifier where this Proxy was created",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				Description: "Name for the Proxy",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_url": {
+				Description: "Destination URL for the Proxy",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"request_reactor_id": {
+				Description: "Request reactor ID for the Proxy",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"require_auth": {
+				Description: "Require auth for the Proxy",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```

Comment:
This should default to `true` to match the behavior of the API, or we have to make it nullable to take the API's default value.

Reply:
when I tested this I got confused between the default bool in Go and the response from the API. good catch